### PR TITLE
#765 Fix request retry check in dao module

### PR DESCRIPTION
**Pull request by GeorgiChochov**

### CODE REVIEW

File: dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala [hunk 1]
```diff
@@ -182,11 +182,11 @@ object EnceladusRestDAO extends EnceladusDAO {
 
         if (isUnauthorized) {
           log.warn(s"Unauthorized GET request for Menas URL: $url")
+          if (retriesLeft == 0) {
```
Comment (Zejnilovic) [inline dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala]: would you mind putting `<= 0` in? Just o be safe if I don't know for what reason we decide in the future to manipulate the `restriesLeft`

Comment (Zejnilovic) [reply dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala]: Maybe add `else log.info("trying to log in")`

File: dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala [hunk 2]
```diff
@@ -182,11 +182,11 @@ object EnceladusRestDAO extends EnceladusDAO {
...
+            throw UnauthorizedException(s"Unable to reauthenticate after retries")
+          }
           log.warn(s"Expired session, reauthenticating")
           if (enceladusLogin()) {
```
Comment (Zejnilovic) [inline dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala]: if the login fails won't it go straight to else on line 193 which will throw an exception that will be caught by line 210 and there will be no retries? Meaning never more than 1 retrie for logging in.

File: dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala (hunk 1 above)
Comment (GeorgiChochov) [reply dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala]: 1. Sure.
2. Else is implied here and it does log it with a warning.

File: dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala (hunk 2 above)
Comment (GeorgiChochov) [reply dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala]: It's intended to only have 1 retry.

**Pull request merged**